docs/topics/development/testing_and_quality_assurance.md

```diff
@@ -0,0 +1,149 @@
+
+- **Ruff**: Used for linting and formatting Python code.
+- **Prettier**: Used for linting and formatting Node.js code.
+
+To format and lint your code, run:
+
+```sh
+make format
+```
+
+This command ensures that your code adheres to the project's style guidelines.
+
+## Build Verification
+
+Build verification is a crucial step in our CI pipeline. We run all CI checks using an actual Docker build that is nearly identical to our local build. This ensures a high level of confidence between what you develop locally and what is tested in CI.
+
+We verify the build itself by running checks against both our Docker configurations and our Docker container. To verify your container locally, run:
+
+```sh
+make check
+```
+
+This command runs a series of checks to ensure the Docker setup and container are functioning correctly.
+
+## Testing
+
+The project primarily uses Python for tests, with some Node.js tests as well. Here's how to handle testing:
+
+- **Python Tests**: The majority of the tests are written in Python.
+- **Node.js Tests**: There are some tests written in Node.js.
+
+To run specialized tests, you can shell into the container or check `Makefile-docker` for specific test commands.
+
+### Running Tests
+
+To run the whole test suite, use:
+
+```sh
+make test
+```
+
+- **Running Specialized Tests**: Shell into the container and run the desired test command.
+
+```sh
+make shell
+make test_some_specific_test
+```
+
+To speed up test execution, you can parallelize and/or share tests. The project uses `pytest-split` and `pytest-xdist` to facilitate this. These tools allow for distributed and parallel test execution, significantly reducing test times.
+
+- **Parallel Testing**:
+
+```sh
+pytest -n auto  # pytest-xdist
+```
+
+#### Configuration
+
+Configuration for your unit tests is handled automatically. Ensure that the database credentials in your settings have full permissions to modify a database with `test_` prepended to it. By default, the database name is `olympia`, so the test database is `test_olympia`.
+
+If the code you are working on is related to search, you'll want to run Elasticsearch tests. Ensure Elasticsearch is installed. See the :ref:`elasticsearch` page for details.
+
+- To exclude Elasticsearch tests:
+
+```sh
+make test_no_es
+```
+
+- To run only Elasticsearch tests:
+
+```sh
+make test_es
+```
+
+### Using `pytest` Directly
+
+For advanced users, you can connect to the Docker container using `make shell` and then use `pytest` directly, which allows for finer-grained control of the test suite.
+
+```sh
+pytest
+```
+
+Examples of running subsets of the test suite:
+
+- `pytest -m es_tests` to run tests marked as `es_tests`.
+- `pytest -k test_no_license` to run tests with `test_no_license` in their name.
+- `pytest src/olympia/addons/tests/test_views.py::TestLicensePage::test_no_license` to run a specific test.
```

Comment:
I don't know how much detail we want to go into, but `pytest src/olympia/addons/tests/test_views.py::TestLicensePage::test_no_license` is the _most_ specific example. 

Personally, I never run `make test` and always end up running tests at a specific level locally, e.g.
`pytest src/olympia/addons/tests/test_views.py::TestLicensePage` for all the tests in the class;
`pytest src/olympia/addons/tests/test_views.py` for all the tests in the file; 
`pytest src/olympia/addons/tests/` for all the addons tests.